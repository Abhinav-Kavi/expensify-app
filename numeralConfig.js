import numeral from 'numeral';

const numeralConfig = (() => {
  numeral.register('locale', 'In', {
    delimiters: {
        thousands: ',',
        decimal: '.'
    },
    abbreviations: {
        thousand: 'k',
        million: 'm',
        billion: 'b',
        trillion: 't'
    },
    ordinal : function (number) {
        return number === 1 ? 'er' : 'ème';
    },
    currency: {
        symbol: '₹'
    }
  });

  // switch between locales
  numeral.locale('In');
})();