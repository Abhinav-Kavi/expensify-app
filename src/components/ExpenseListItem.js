import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';


const ExpenseListItem = ({description,amount,createdAt,id})=>(
  <div>
    <Link to={`/edit/${id}`}> <h3>{description}</h3> </Link>
    <p>
     {numeralConfig()}
     {numeral(amount).format('$0,0.00')} 
     -
     {moment(createdAt).format('MMMM Do, YYYY')}</p>   
  </div>
);

const numeralConfig = () => {
  numeral.register('locale', 'In', {
    delimiters: {
        thousands: ',',
        decimal: '.'
    },
    abbreviations: {
        thousand: 'k',
        million: 'm',
        billion: 'b',
        trillion: 't'
    },
    ordinal : function (number) {
        return number === 1 ? 'er' : 'ème';
    },
    currency: {
        symbol: '₹'
    }
  });

  // switch between locales
  numeral.locale('In');
}

export default ExpenseListItem;