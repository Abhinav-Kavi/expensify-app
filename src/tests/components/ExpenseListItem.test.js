import React from 'react';
import {shallow} from 'enzyme';
import expenses from '../fixtures/expenses';
import ExpenseListItem from '../../components/ExpenseListItem';
import '../../../numeralConfig';

test('should render ExpenseListItem with provided expense',()=>{
  const wrapper = shallow(<ExpenseListItem  {...expenses[0]} />);
  expect(wrapper).toMatchSnapshot();
});