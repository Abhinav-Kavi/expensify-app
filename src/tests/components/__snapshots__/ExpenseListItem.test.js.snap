// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with provided expense 1`] = `
<Link
  className="list-item"
  replace={false}
  to="/edit/1"
>
  <div>
    <h3
      className="list-item__title"
    >
      Credit Card Bill
    </h3>
    <span
      className="list-item__sub-title"
    >
      December 28th, 1969
    </span>
  </div>
  <h3
    className="list-item__data"
  >
    â‚¹10,000.00
  </h3>
</Link>
`;
